thinc/backends/_custom_kernels.py

```diff
@@ -55,12 +55,30 @@ def compile_mmh(src):
 hash_data_kernel = compile_mmh(MMH_SRC)
 
 
-def seq2col(X, nW, out=None, threads_per_block=128, num_blocks=128):
-    if out is None:
-        out = cupy.zeros((X.shape[0], X.shape[1] * ((nW * 2) + 1)), dtype="f")
+def check_seq2col_lens(lens, B):
+    if lens is None:
+        lens = cupy.array([B], dtype="int32")
+    else:
+        lens = lens.astype("int32")
+        assert cupy.all(lens >= 0), "All sequence lengths must be >= 0"
+        assert cupy.sum(lens) == B, "The lengths must sum up to the batch length"
+
+    return lens
+
+
+def seq2col(X, nW, lens=None, out=None, threads_per_block=128, num_blocks=128):
```

Comment:
Let's improve this a little and make these keyword-only. Otherwise technically inserting the keyword argument first is breaking, which is 😬 

```suggestion
def seq2col(X, nW, *, lens=None, out=None, threads_per_block=128, num_blocks=128):
```